src/components/sources/sourcesContext.js

```diff
@@ -1,115 +1,150 @@
 import { apiTypes } from '../../constants/apiConstants';
 import { helpers } from '../../common';
 import { translate } from '../i18n/i18n';
+import { useConfirmation } from '../../hooks/useConfirmation';
 
 /**
- * On Delete confirmation, and action.
+ * Sources action, onDelete.
  *
  * @param {object} options
- * @param {Function} options.deleteSource
+ * @param {Function} options.deleteSources
  * @param {Function} options.t
+ * @param {Function} options.useConfirmation
  * @param {Function} options.useDispatch
- * @param {Function} options.useSelector
  * @param {Function} options.useSelectorsResponse
- * @returns {Function}
+ * @returns {(function(*): void)|*}
  */
 const useOnDelete = ({
-  deleteSource = reduxActions.sources.deleteSource,
+  deleteSources = reduxActions.sources.deleteSource,
   t = translate,
+  useConfirmation: useAliasConfirmation = useConfirmation,
   useDispatch: useAliasDispatch = storeHooks.reactRedux.useDispatch,
-  useSelector: useAliasSelector = storeHooks.reactRedux.useSelector,
   useSelectorsResponse: useAliasSelectorsResponse = storeHooks.reactRedux.useSelectorsResponse
 } = {}) => {
+  const onConfirmation = useAliasConfirmation();
+  const [sourcesToDelete, setSourcesToDelete] = useState([]);
   const dispatch = useAliasDispatch();
-  const sourceToDelete = useAliasSelector(({ sources }) => sources?.confirmDelete?.source, {});
-  const { error, fulfilled, message } = useAliasSelectorsResponse(({ sources }) => sources?.deleted);
-  const { [apiTypes.API_RESPONSE_SOURCE_ID]: sourceId, [apiTypes.API_RESPONSE_SOURCE_NAME]: sourceName } =
-    sourceToDelete;
+  const {
+    data,
+    error: deletedError,
+    fulfilled: deletedFulfilled
+  } = useAliasSelectorsResponse(({ sources }) => sources?.deleted);
+  const { errorMessage } = data?.[0] || {};
+
+  useShallowCompareEffect(() => {
+    if (sourcesToDelete.length) {
+      const sourceIds = sourcesToDelete.map(source => source[apiTypes.API_RESPONSE_SOURCE_ID]);
+      deleteSources(sourceIds)(dispatch);
+    }
+  }, [sourcesToDelete, deleteSources, dispatch]);
 
-  useEffect(() => {
-    if (sourceId) {
+  useShallowCompareEffect(() => {
+    if (deletedFulfilled && sourcesToDelete.length) {
+      const sourceNames = sourcesToDelete.map(source => source[apiTypes.API_RESPONSE_SOURCE_NAME]);
       dispatch([
         {
-          type: reduxTypes.confirmationModal.CONFIRMATION_MODAL_HIDE
+          type: reduxTypes.toastNotifications.TOAST_ADD,
+          alertType: AlertVariant.success,
+          header: t('toast-notifications.title', {
+            context: ['deleted-source'],
+            count: sourceNames.length
+          }),
+          message: t('toast-notifications.description', {
+            context: ['deleted-source'],
+            name: sourceNames[0],
+            count: sourceNames.length
+          })
+        },
+        {
+          type: reduxTypes.sources.DESELECT_SOURCE,
+          item: sourcesToDelete
```

Comment:
required:
looks like we're missing a piece in the sourcesReducer... add into the reducer the ability to parse an Array of sources to delete